workflow/controller/operator_concurrency_test.go

```diff
@@ -176,9 +176,7 @@ func TestSemaphoreTmplLevel(t *testing.T) {
 
 		// Release the lock
 		woc.operate()
-		assert.NotNil(t, woc.wf.Status.Synchronization)
-		assert.NotNil(t, woc.wf.Status.Synchronization.Semaphore)
-		assert.Equal(t, 0, len(woc.wf.Status.Synchronization.Semaphore.Holding[0].Holders))
+		assert.Nil(t, woc.wf.Status.Synchronization)
```

Comment:
do we need to new tests else where now this test does not check these conditions?

Reply:
The above change is testing the correct usecase.  Release all is happening inside operate a function it will be nill if it released all locks.